### ISSUE TITLE
Incorrect "Try with resources only supports variable declarations" problem

### ISSUE THREAD

**Issue by ctubbsii** [javaparser/javaparser#2820]
I have valid Java 11 code that parses unsuccessfully and generates one problem:

```
(line 114,col 7) Try with resources only supports variable declarations.
```

Note: line 114 is the `try (client)` line below:

```java
  public static Cleanable batchWriterAndClientCloser(Object o, Logger log, BatchWriter bw,
      AccumuloClient client) {
    requireNonNull(log);
    requireNonNull(bw);
    requireNonNull(client);
    return CLEANER.register(o, () -> {
      try (client) {
        bw.close();
      } catch (MutationsRejectedException e) {
        log.error("Failed to close BatchWriter; some mutations may not be applied", e);
      }
    });
  }
```

This problem occurs most recently with version 3.16.1, but also seems to affect older versions as well.

The reason this is valid Java code, and the parser Problem is wrong is because of https://bugs.openjdk.java.net/browse/JDK-7196163 where Java 9 and later now allow the use of a final, or effectively final, variable as a resource, without needing to declare an additional variable.

**Comment by jlerbsc** [comment]
Hi @ctubbsii Did you set the language level in ParserConfiguration?

**Comment by ctubbsii** [comment]
> Hi @ctubbsii Did you set the language level in ParserConfiguration?

No, I did not. Thank you for the tip. I will close this issue, as it does not occur when the language level is set correctly. I will need to make my own code determine the language level before it instantiates the parser.

**Comment by jlerbsc** [comment]
JAVA8 is used by default.

**Comment by ctubbsii** [comment]
Thanks @jlerbsc ; I was able to fix my issue in revelc/impsort-maven-plugin#37
